openvm/src/plonk/air_to_plonkish.rs

```diff
@@ -75,12 +75,13 @@ where
         &mut self,
         algebraic_expr: &AlgebraicExpression<T>,
         assert_zero: bool,
-    ) -> Variable<AlgebraicReference> {
-        if let Some(var) = self.cache.get(algebraic_expr) {
```

Comment:
I think this is a bug (also on `main`): It ignores `assert_zero`. So if you happened to have computed the expression before and then assert it's zero, it does nothing.